### ISSUE TITLE
stack working on arm64 cluster

### ISSUE THREAD

**Issue by lumir-mrkva** [swarmpit/swarmpit#431]

**Comment by lumir-mrkva** [comment]
not looking good so far with java on *aarch64*

```
compit_app.1.a6stwg5w7hsn@swarmpit    | Caused by: java.lang.UnsatisfiedLinkError: java.lang.UnsatisfiedLinkError: /tmp/jffi8950710721863176097.so: Error loading shared library ld-linux-aarch64.so.1: No such file or directory (needed by /tmp/jffi8950710721863176097.so)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at java.lang.ClassLoader$NativeLibrary.load(Native Method)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1941)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1824)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at java.lang.Runtime.load0(Runtime.java:809)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at java.lang.System.load(System.java:1086)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at com.kenai.jffi.internal.StubLoader.loadFromJar(StubLoader.java:371)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at com.kenai.jffi.internal.StubLoader.load(StubLoader.java:258)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at com.kenai.jffi.internal.StubLoader.<clinit>(StubLoader.java:444)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at java.lang.Class.forName0(Native Method)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at java.lang.Class.forName(Class.java:348)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at com.kenai.jffi.Init.load(Init.java:68)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at com.kenai.jffi.Foreign$InstanceHolder.getInstanceHolder(Foreign.java:49)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at com.kenai.jffi.Foreign$InstanceHolder.<clinit>(Foreign.java:45)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at com.kenai.jffi.Foreign.getInstance(Foreign.java:103)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at com.kenai.jffi.Type$Builtin.lookupTypeInfo(Type.java:242)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at com.kenai.jffi.Type$Builtin.getTypeInfo(Type.java:237)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at com.kenai.jffi.Type.resolveSize(Type.java:155)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at com.kenai.jffi.Type.size(Type.java:138)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at jnr.ffi.provider.jffi.NativeRuntime$TypeDelegate.size(NativeRuntime.java:187)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at jnr.ffi.provider.AbstractRuntime.<init>(AbstractRuntime.java:48)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at jnr.ffi.provider.jffi.NativeRuntime.<init>(NativeRuntime.java:66)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at jnr.ffi.provider.jffi.NativeRuntime.<init>(NativeRuntime.java:41)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at jnr.ffi.provider.jffi.NativeRuntime$SingletonHolder.<clinit>(NativeRuntime.java:62)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at jnr.ffi.provider.jffi.NativeRuntime.getInstance(NativeRuntime.java:58)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at jnr.ffi.provider.jffi.Provider.<init>(Provider.java:29)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at java.lang.Class.newInstance(Class.java:442)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at jnr.ffi.provider.FFIProvider$SystemProviderSingletonHolder.getInstance(FFIProvider.java:68)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at jnr.ffi.provider.FFIProvider$SystemProviderSingletonHolder.<clinit>(FFIProvider.java:57)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at jnr.ffi.provider.FFIProvider.getSystemProvider(FFIProvider.java:35)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at jnr.ffi.LibraryLoader.create(LibraryLoader.java:73)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at jnr.unixsocket.Native.<clinit>(Native.java:76)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at jnr.unixsocket.UnixSocketChannel.<init>(UnixSocketChannel.java:101)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at jnr.unixsocket.UnixSocketChannel.open(UnixSocketChannel.java:60)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at swarmpit.socket.HttpUnixSocket.<init>(HttpUnixSocket.java:30)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at swarmpit.socket.UnixSocketFactory.createSocket(UnixSocketFactory.java:29)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:119)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at org.apache.http.impl.conn.BasicHttpClientConnectionManager.connect(BasicHttpClientConnectionManager.java:325)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:381)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:237)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:185)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:111)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.core$request.invokeStatic(core.clj:496)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.core$request.invoke(core.clj:427)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.core$request.invokeStatic(core.clj:428)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.core$request.invoke(core.clj:427)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clojure.lang.Var.invoke(Var.java:384)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$wrap_request_timing$fn__5898.invoke(client.clj:1078)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$wrap_async_pooling$fn__5906.invoke(client.clj:1111)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.headers$wrap_header_map$fn__4845.invoke(headers.clj:147)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$wrap_query_params$fn__5788.invoke(client.clj:812)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$wrap_basic_auth$fn__5794.invoke(client.clj:835)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$wrap_oauth$fn__5799.invoke(client.clj:852)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$wrap_user_info$fn__5808.invoke(client.clj:872)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$wrap_url$fn__5880.invoke(client.clj:1030)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$wrap_decompression$fn__5597.invoke(client.clj:416)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$wrap_input_coercion$fn__5712.invoke(client.clj:632)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$wrap_additional_header_parsing$fn__5737.invoke(client.clj:687)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$wrap_output_coercion$fn__5699.invoke(client.clj:576)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$wrap_exceptions$fn__5550.invoke(client.clj:250)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$wrap_accept$fn__5752.invoke(client.clj:730)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$wrap_accept_encoding$fn__5759.invoke(client.clj:752)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$wrap_content_type$fn__5746.invoke(client.clj:713)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$wrap_form_params$fn__5845.invoke(client.clj:954)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$wrap_nested_params$fn__5866.invoke(client.clj:988)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$wrap_flatten_nested_params$fn__5875.invoke(client.clj:1012)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$wrap_method$fn__5813.invoke(client.clj:888)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.cookies$wrap_cookies$fn__4058.invoke(cookies.clj:131)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.links$wrap_links$fn__5158.invoke(links.clj:63)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$wrap_unknown_host$fn__5883.invoke(client.clj:1041)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$request_STAR_.invokeStatic(client.clj:1214)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$request_STAR_.invoke(client.clj:1207)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$get.invokeStatic(client.clj:1220)
compit_app.1.a6stwg5w7hsn@swarmpit    |         at clj_http.client$get.doInvoke(client.clj:1216)

```

after installing libc6-compat
```
swarmpit_app.1.bm8duu2knlmz@swarmpit    | Dec 12, 2019 12:44:52 AM swarmpit.database invoke
swarmpit_app.1.bm8duu2knlmz@swarmpit    | INFO: ... connected after 0 sec
swarmpit_app.1.bm8duu2knlmz@swarmpit    | Dec 12, 2019 12:44:52 AM swarmpit.database invoke
swarmpit_app.1.bm8duu2knlmz@swarmpit    | INFO: Waiting for InfluxDB ...
swarmpit_app.1.bm8duu2knlmz@swarmpit    | Dec 12, 2019 12:44:52 AM swarmpit.database invoke
swarmpit_app.1.bm8duu2knlmz@swarmpit    | INFO: ... connected after 0 sec
swarmpit_app.1.bm8duu2knlmz@swarmpit    | Dec 12, 2019 12:44:52 AM swarmpit.server invoke
swarmpit_app.1.bm8duu2knlmz@swarmpit    | INFO: Swarmpit running on port 8080
swarmpit_app.1.bm8duu2knlmz@swarmpit    | #
swarmpit_app.1.bm8duu2knlmz@swarmpit    | # A fatal error has been detected by the Java Runtime Environment:
swarmpit_app.1.bm8duu2knlmz@swarmpit    | #
swarmpit_app.1.bm8duu2knlmz@swarmpit    | #  SIGSEGV (0xb) at pc=0x0000000000005e30, pid=1, tid=0x0000ffff85175aa0
swarmpit_app.1.bm8duu2knlmz@swarmpit    | #
swarmpit_app.1.bm8duu2knlmz@swarmpit    | # JRE version: OpenJDK Runtime Environment (8.0_212-b04) (build 1.8.0_212-b04)
swarmpit_app.1.bm8duu2knlmz@swarmpit    | # Java VM: OpenJDK 64-Bit Server VM (25.212-b04 mixed mode linux-aarch64 compressed oops)
swarmpit_app.1.bm8duu2knlmz@swarmpit    | # Derivative: IcedTea 3.12.0
swarmpit_app.1.bm8duu2knlmz@swarmpit    | # Distribution: Custom build (Sat May  4 17:44:26 UTC 2019)
swarmpit_app.1.bm8duu2knlmz@swarmpit    | # Problematic frame:
swarmpit_app.1.bm8duu2knlmz@swarmpit    | # C  0x0000000000005e30
swarmpit_app.1.bm8duu2knlmz@swarmpit    | #
swarmpit_app.1.bm8duu2knlmz@swarmpit    | # Core dump written. Default location: /usr/src/app/core or core.1
swarmpit_app.1.bm8duu2knlmz@swarmpit    | #
swarmpit_app.1.bm8duu2knlmz@swarmpit    | # An error report file with more information is saved as:
swarmpit_app.1.bm8duu2knlmz@swarmpit    | # /usr/src/app/hs_err_pid1.log
swarmpit_app.1.bm8duu2knlmz@swarmpit    | #
swarmpit_app.1.bm8duu2knlmz@swarmpit    | # If you would like to submit a bug report, please include
swarmpit_app.1.bm8duu2knlmz@swarmpit    | # instructions on how to reproduce the bug and visit:
swarmpit_app.1.bm8duu2knlmz@swarmpit    | #   https://icedtea.classpath.org/bugzilla
swarmpit_app.1.bm8duu2knlmz@swarmpit    | #
```

**Comment by lumir-mrkva** [comment]
different image for couch db is needed

**Comment by lumir-mrkva** [comment]
hmmm... mby just ammend socket factory

```swarmpit_app.1.rlwolf04nr82@arm64    | Exception in thread "main" java.lang.UnsatisfiedLinkError: could not load FFI provider jnr.ffi.provider.jffi.Provider
swarmpit_app.1.rlwolf04nr82@arm64    | 	at jnr.ffi.provider.InvalidProvider$1.loadLibrary(InvalidProvider.java:48)
swarmpit_app.1.rlwolf04nr82@arm64    | 	at jnr.ffi.LibraryLoader.load(LibraryLoader.java:325)
swarmpit_app.1.rlwolf04nr82@arm64    | 	at jnr.unixsocket.Native.<clinit>(Native.java:80)
swarmpit_app.1.rlwolf04nr82@arm64    | 	at jnr.unixsocket.UnixSocketChannel.<init>(UnixSocketChannel.java:101)
swarmpit_app.1.rlwolf04nr82@arm64    | 	at jnr.unixsocket.UnixSocketChannel.open(UnixSocketChannel.java:60)
swarmpit_app.1.rlwolf04nr82@arm64    | 	at swarmpit.socket.HttpUnixSocket.<init>(HttpUnixSocket.java:30)
swarmpit_app.1.rlwolf04nr82@arm64    | 	at swarmpit.socket.UnixSocketFactory.createSocket(UnixSocketFactory.java:29)
swarmpit_app.1.rlwolf04nr82@arm64    | 	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:119)
swarmpit_app.1.rlwolf04nr82@arm64    | 	at org.apache.http.impl.conn.BasicHttpClientConnectionManager.connect(BasicHttpClientConnectionManager.java:325)
swarmpit_app.1.rlwolf04nr82@arm64    | 	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:381)
swarmpit_app.1.rlwolf04nr82@arm64    | 	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:237)
swarmpit_app.1.rlwolf04nr82@arm64    | 	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:185)
swarmpit_app.1.rlwolf04nr82@arm64    | 	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
swarmpit_app.1.rlwolf04nr82@arm64    | 	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:111)
swarmpit_app.1.rlwolf04nr82@arm64    | 	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
swarmpit_app.1.rlwolf04nr82@arm64    | 	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
```

**Comment by nohaapav** [comment]
This is done i guess.

**Comment by BobTheDestroyer** [comment]
Bump this one: 
treehouses/couchdb:2.3.0 does not seem to support arch64 for now:

"No such image: treehouses/couchdb:2.3.0@sha256:ae73732dfa8da686073e092636dadcdf7f691cce8b586f05cd520f715ca49235"

When I tried to pull the image directly from treehouses/couchdb:2.3.0:

bob@MasterPine:~$ docker pull treehouses/couchdb:2.3.0
2.3.0: Pulling from treehouses/couchdb
no matching manifest for linux/arm64/v8 in the manifest list entries

checked it out, version 2.3.1 does work with arm64, on top of arm and amd64.

When I modified the docker-compose.arm.yml to use 2.3.1 , the service can now function and the app can go live.

New problem: all services are up and healthcheck works:

`
bob@MasterPine:~$ docker service ls
ID                  NAME                MODE                REPLICAS            IMAGE                      PORTS
j6021cvxoiqw        swarmpit_agent      global              3/3                 swarmpit/agent:latest
rli942q77zmq        swarmpit_app        replicated          1/1                 swarmpit/swarmpit:latest   *:888->8080/tcp
s8s8yvg7uejh        swarmpit_db         replicated          1/1                 treehouses/couchdb:2.3.1
tt5nm8o6q5u2        swarmpit_influxdb   replicated          1/1                 influxdb:1.7
`

via wget can access an index.html, however via curl or web browser nothing happens.
